File: mobile/shared/src/commonMain/kotlin/com/preuni/shared/domain/user/Student.kt
package com.preuni.shared.domain.user

data class Student(
    val id: String,
    val displayName: String,
    val username: String,
    val email: String,
    val avatarUrl: String?,
    val xpTotal: Long,
    val streakCount: Int,
    val readinessScore: Double,
    val onboardingCompleted: Boolean,
)

File: mobile/src/types/student.ts
import { z } from 'zod';

export const StudentSchema = z.object({
  id: z.string().uuid(),
  email: z.string().email(),
  displayName: z.string(),
  username: z.string().nullable().optional(),
  avatarUrl: z.string().url().nullable().optional(),
  xpTotal: z.number().nonnegative(),
  streakCount: z.number().int().nonnegative(),
  readinessScore: z.number().min(0).max(100),
  onboardingCompleted: z.boolean(),
  emailVerified: z.boolean().optional(),
});

export type Student = z.infer<typeof StudentSchema>;
